sql/hive-thriftserver/v2.3/src/main/java/org/apache/hive/service/server/HiveServer2.java

```diff
@@ -158,25 +156,16 @@ private static void startHiveServer2() throws Throwable {
 
   public static void main(String[] args) {
     HiveConf.setLoadHiveServer2Config(true);
-    try {
-      ServerOptionsProcessor oproc = new ServerOptionsProcessor("hiveserver2");
-      ServerOptionsProcessorResponse oprocResponse = oproc.parse(args);
+    ServerOptionsProcessor oproc = new ServerOptionsProcessor("hiveserver2");
+    ServerOptionsProcessorResponse oprocResponse = oproc.parse(args);
 
-      // NOTE: It is critical to do this here so that log4j is reinitialized
-      // before any of the other core hive classes are loaded
-      String initLog4jMessage = LogUtils.initHiveLog4j();
-      LOG.debug(initLog4jMessage);
-      HiveStringUtils.startupShutdownMessage(HiveServer2.class, args, LOG);
+    HiveStringUtils.startupShutdownMessage(HiveServer2.class, args, LOG);
 
-      // Log debug message from "oproc" after log4j initialize properly
-      LOG.debug(oproc.getDebugMessage().toString());
+    // Log debug message from "oproc" after log4j initialize properly
+    LOG.debug(oproc.getDebugMessage().toString());
```

Comment:
is it OK to call `LOG.debug` without initialization?

Reply:
It will ignore the log message, but not throw any exception. As we see here, we initialise spark loggers after this call, https://github.com/apache/spark/blob/master/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala#L87 
So calling logger before this (even using commons.logging) will not print anything in logs as it doesn't have any appenders

Reply:
the toString() could trigger exceptions (NPE etc) If you aren't careful and there is the extra cost of that operation. Its why scala string expansion and SLF4J {} entries are better

Reply:
shall we just remove this line if it's noop?

Reply:
@steveloughran as i see the code `toString()` is over a `StringBuilder` which is always initialised in `org.apache.hive.service.server.HiveServer2.ServerOptionsProcessor#debugMessage` and as also this is java file dint use scala string expansion. Please correct me if wrong.

@cloud-fan removed as per suggestion